### PR TITLE
Feature/75 matrix modules

### PR DESCRIPTION
**Pull request by Pete-Robelou**
I've created a `matrix` CSS component which works in conjunction with a bunch of utility classes to handle this.

It's possible we may want to look at wrapping up the utility classes into part of the `matrix` CSS but let's assess this further into the build.

### CODE REVIEW

File: src/css/component.matrix.css [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  text-decoration: underline;
+}
+
+@media (--mq-to-lg) {
+  .matrix {
+    display: flex;
+    margin-left: calc(-1 * var(--matrix-gutters-x));
+    margin-right: calc(-1 * var(--matrix-gutters-x));
+    overflow-x: auto;
+    scroll-snap-type: x mandatory;
```
Comment (adamduncan) [inline src/css/component.matrix.css:51]: Tasty progressive enhancement 🤙

File: src/css/component.matrix.css [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  .matrix__list > li:nth-last-child(4):first-child {
+    grid-column: 1 / span 2;
+    grid-row: 1 / span 2;
+  }
+
+  .matrix__list > li:nth-last-child(4):first-child ~ :nth-child(2) {
+    grid-column: 3 / span 2;
+  }
+
+  .matrix__list > li:nth-last-child(4):first-child ~ :nth-child(3) {
```
Comment (adamduncan) [inline src/css/component.matrix.css:94]: Just a note: Might be one to document to ensure that content editors understand the variants that exist, and how it behaves with differening amounts of items. (Are we shooting to support variations with just 3 and 4 items for now?)

File: src/css/utilities.typography.css [hunk 3]
```diff
@@ -56,6 +56,14 @@
   line-height: var(--font-leading-normal);
 }
 
+.text-leading-8 {
```
Comment (adamduncan) [inline src/css/utilities.typography.css:59]: Curious about these leading-8/10s. Is this mixing the concept of _size_ and _type_? What scenarios might we need this over our baseline definition of line-heights?

**Review by adamduncan — APPROVED** [review]
Looking good, @Pete-Robelou 👍

File: src/css/component.matrix.css (hunk 2 above)
Comment (Pete-Robelou) [reply src/css/component.matrix.css:94]: Yeah, I've based this on the designs. I only saw a 3 and 4 layout.

It's a relatively easy fix to add CSS for more or less.

I can add comments to the README's for pages that use matrix modules defining the use of 3 or 4 modules.

File: src/css/utilities.typography.css (hunk 3 above)
Comment (Pete-Robelou) [reply src/css/utilities.typography.css:59]: Ah, these were supposed to be removed as they're not needed.

**Pull request merged**